airgun/entities/host.py

```diff
@@ -218,18 +218,24 @@ def get_webconsole_content(self, entity_name):
         view = self.navigate_to(self, 'Details', entity_name=entity_name)
         view.webconsole.click()
         view.validations.assert_no_errors()
+        # switch to the last opened tab,
+        self.browser.switch_to_window(self.browser.window_handles[-1])
```

Comment:
This is dangerous behavior to be run directly without some mechanism to mitigate failures to do anything between when you switch tot he window and when you switch back to window handle 0.

If there are any failures in any of the commands in between, the browser needs a mechanism to get back to the original tab.

I would at a minimum put the behavior into a try/except block with a generic exception catch, so that the browser can be returned to the original window handle.

Providing a mechanism within the library, not just within this method, is something we need to consider for core airgun refactoring efforts.

Reply:
@tstrych pointed out this is a backport of previously accepted behavior.